aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -261,6 +269,19 @@ module aptos_framework::account {
 
         // Update the account with the new authentication key
         account_resource.authentication_key = new_auth_key;
+
+        // Emit event upon key rotation
+        let account = borrow_global_mut<Account>(addr);
+        event::emit_event<KeyRotationEvent>(
+            &mut account.key_rotation_events,
+            KeyRotationEvent {
+                // Address of rotator may be different from the addr of the account
+                // once we allow key rotation on behalf of another account
+                address_of_rotator: addr,
```

Comment:
address_of_rotator can be found from ledger info which is the same as `account: &signer`, no need to store it.